crypto/ethsecp256k1/ethsecp256k1.go

```diff
@@ -103,33 +141,61 @@ func (key PubKey) Address() tmcrypto.Address {
 }
 
 // Bytes returns the raw bytes of the ECDSA public key.
-// The function panics if the key cannot be marshaled to bytes.
-func (key PubKey) Bytes() []byte {
-	bz, err := CryptoCodec.MarshalBinaryBare(key)
-	if err != nil {
-		panic(err)
+func (pubKey PubKey) Bytes() []byte {
+	return pubKey.Key
+}
+
+// String implements the fmt.Stringer interface.
+func (pubKey *PubKey) String() string {
+	return fmt.Sprintf("EthPubKeySecp256k1{%X}", pubKey.Key)
+}
+
+// Type returns eth_secp256k1
+func (pubKey *PubKey) Type() string {
+	return KeyType
+}
+
+// Equals returns true if the pubkey type is the same and their bytes are deeply equal.
+func (pubKey *PubKey) Equals(other cryptotypes.PubKey) bool {
+	return pubKey.Type() == other.Type() && bytes.Equal(pubKey.Bytes(), other.Bytes())
+}
+
+// MarshalAmino overrides Amino binary marshalling.
+func (pubKey PubKey) MarshalAmino() ([]byte, error) {
+	return pubKey.Key, nil
+}
+
+// UnmarshalAmino overrides Amino binary marshalling.
+func (pubKey *PubKey) UnmarshalAmino(bz []byte) error {
+	if len(bz) != PubKeySize {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidPubKey, "invalid pubkey size, expected %d, got %d", PubKeySize, len(bz))
 	}
-	return bz
+	pubKey.Key = bz
+
+	return nil
 }
 
-// VerifyBytes verifies that the ECDSA public key created a given signature over
+// MarshalAminoJSON overrides Amino JSON marshalling.
+func (pubKey PubKey) MarshalAminoJSON() ([]byte, error) {
+	// When we marshal to Amino JSON, we don't marshal the "key" field itself,
+	// just its contents (i.e. the key bytes).
+	return pubKey.MarshalAmino()
+}
+
+// UnmarshalAminoJSON overrides Amino JSON marshalling.
+func (pubKey *PubKey) UnmarshalAminoJSON(bz []byte) error {
+	return pubKey.UnmarshalAmino(bz)
+}
+
+// VerifySignature verifies that the ECDSA public key created a given signature over
 // the provided message. It will calculate the Keccak256 hash of the message
 // prior to verification.
-func (key PubKey) VerifyBytes(msg []byte, sig []byte) bool {
+func (pubKey PubKey) VerifySignature(msg []byte, sig []byte) bool {
 	if len(sig) == 65 {
```

Comment:
maybe create a const for 65 as signature length ?

Reply:
sig len is 64, the extra byte is for recovery ID

Reply:
i think @freddyli7  was asking to make the signature length to a constant variable instead of it being a magic number here. but maybe im wrong haha